### ISSUE TITLE
Allow specifying "orchestrationClient" instead of "durableClient" in function.json (v2.x)

### ISSUE THREAD

**Issue by cgillum** [Azure/azure-functions-durable-extension#1290]
## Problem statement
We made a breaking change from Durable Functions v1 to v2 by renaming the [durable client binding](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-bindings#orchestration-client) from `orchestrationClient` to `durableClient`. This is fine, and we documented the change, but we recently decided that we want to make it easier for customers to upgrade from v1.x to v2.x.

For JavaScript customers specifically, the only change they need to make is to update their orchestration client bindings in their **function.json** files.  Ideally we can remove this requirement to make the transition much smoother. Without this backwards compatibility, it could be hard for them to figure out why things aren't working when they upgrade their extension bundles from v1 to v2.

## Proposed solution
I would like it if we could support both "durableClient" **or** "orchestrationClient" in functions.json. It might be as simple as defining a new public `OrchestrationClientAttribute` class in the .NET extension and having it derive from `DurableClientAttribute`. Perhaps something like this:

```csharp
[Binding]
[AttributeUsage(AttributeTargets.Parameter)]
[Obsolete($"{nameof(OrchestrationClientAttribute)} is obsolete. Use {nameof(DurableClientAttribute)} instead.")]
public class OrchestrationClientAttribute : DurableClientAttribute
{
}
```

A few notes:
* We want to mark this as `[Obsolete]` so that C# developers don't try to use it.
* We will need to remove the `sealed` part of the `DurableClientAttribute` declaration or else the C# compiler won't let us derive from that base attrbute.
* I don't think anyone has every tried this before (deriving from a base class attribute) and I'm not actually sure if this will even work. We'll need to test it. If it doesn't work, we probably will need to define two unrelated attributes that are treated the same everywhere in code.

If successful, we should be able to take a JavaScript sample, replace `"type":"durableClient"` with `"type":"orchestrationClient"` in function.json and have everything still work.

**Comment by cgillum** [comment]
@davidmrdavid this might be a good work item for you to tackle for the next extension release. What do you think? @ConnorMcMahon should be able to provide some guidance as well.

FYI @anthonychu

**Comment by ConnorMcMahon** [comment]
I think we have tried base class attributes before and ran into some issues. Unfortunately, I don't remember those issues, so it's possible they don't apply here.

The other option is we just create a whole new attribute that is functionally just a copy of the current `DurableClientAttribute`, and set up the same exact binding logic.

**Comment by davidmrdavid** [comment]
Thanks for the suggestion, Chris. I'll happily take over this ASAP. 
So we're essentially adding a backwards compability layer, makes sense. Perhaps we should also consider giving users a warning noting that, after some time, we may drop this backwards compability layer.

In any case, I'll investigate this further and give updates soon 👍

**Comment by cgillum** [comment]
Good point about the deprecation warning. Maybe we could write a warning message in their logs if we see they’re using the obsolete binding name? I’m open to ideas about how we do this.

**Comment by davidmrdavid** [comment]
@cgillum 

Yup! I'm thinking we can log a red warning every time the function with the outdated binding executes, perhaps saying something like:

`[WARNING]: Binding type 'orchestrationClient' is deprecated in Functions V2, please update to 'durableClient' in './DurableTrigger/functions.json`

Furthermore, if we want to provide the best experience, I think this should prompt VSCode to raise a notification saying something like "looks like your `function.json` uses a deprecated binding..upgrade automatically?` And then a click should enable the VSCode extension to do the replacing for us.

Now that would be *really* convenient :smile:

**Comment by cgillum** [comment]
A VS notification would indeed be a more visible way to communicate this (at least at development time). I wonder, though, if it's really worth the extra effort to add that to the Functions VS code extension. I feel like warning text + a manual edit might be good enough, but I'm open to discussion.

**Comment by davidmrdavid** [comment]
I agree with your intuition here, let's start with just the warning text + manual edit then

**Comment by davidmrdavid** [comment]
Merged the C# changes.
Still pending is my stretch goal of VSCode warnings

**Comment by ConnorMcMahon** [comment]
Closed with #1295
